### DISCUSSION TITLE
Move whole payload to other topic

### DISCUSSION THREAD

**Discussion by Clockwork-Muse** [nanomq/nanomq#1836] [Q&A]
I'm writing an Azure IoT Edge application consisting of several modules.  The Azure team seems to have discontinued their simulator, at least partially, so I'm looking at mocking at least some functionality with a broker. 

One of the things I need to do is republish messages to emulate the usual message-flow mapping of a deployed set of modules.  I've gotten this working, but:
```conf
rules.repub {
  rules = [
    {
      server = "mqtt-tcp://localhost:1883"   # MQTT server address for republishing
      topic = "devices/devcontainer/modules/Processor/messages/scanEvent/%24.ce=utf-8&%24.ct=application%2Fjson&%24.on=output1/" # Topic to republish on
      proto_ver = 5                          # MQTT protocol version
      clientid = "ProcessingModule"          # Client ID for the republishing client
      sql = "SELECT payload.id as id, payload.source as source, payload.type as type, payload.data as data, payload.time as time, payload.specversion as specversion, payload.datacontenttype as datacontenttype FROM \"devices/devcontainer/modules/BarcodeReader/messages/events/#\"" # SQL clause for the rule
    }
 ]
}
```

_this works_, but it's annoying to remap ever single subproperty of `payload`.

Is there any other/simpler way to remap the entirety of `payload` so that the other topic sees the "same" message?  I tried `payload.*`, but that doesn't work.

<hr />

One of my other options is looking at migrating to something that also implements AMQP, since that's the default protocol for the edge hub, but this is what I have working at the moment...

**Comment by JaylinYu** [comment]
Rule Engine is not under actively development now. We don't have much time to support variety of SQL languages.
 My advice is ... why don't u have a look at bridging?

**Reply by Clockwork-Muse** [reply]
Hmm...
I originally thought that I might need the rule engine because of the format of the remapping, but it turns out to not be as bad as I thought.  I guess I probably _can_ get away with self-bridging...
